doc/api/url.md

```diff
@@ -921,6 +921,51 @@ console.log(url.format(myURL, { fragment: false, unicode: true, auth: false }));
 // Prints 'https://你好你好/?abc'
 ```
 
+## File URL Utility Functions
+
+When working with `file:///` URLs in Node.js (eg when working with ES modules
+which are keyed in the registry by File URL), the utility functions
+`urlToFilePath` and `fileUrlToPath` are provided to convert to and from file
+paths.
+
+The edge cases handled by these functions include percent-encoding and decoding
+as well as cross-platform support.
+
+For example, instead of writing:
+
+```js
+// BAD:
+// - fails in Windows
+// - doesn't handle loading paths using extended non-latin characters
```

Comment:
not just that, also things like `" "` which are percent encoded

Reply:
This isn't meant to be comprehensive, but we can certainly pad out the examples.

I expect this will also change over time, so I don't think we want to keep the docs exactly in sync with the code here either.

Reply:
agree, but examples seem more prudent than statements like this which could be taken as the only exceptions thereby leading people only on unix and in latin charsets to feel this API is not needed